### PR TITLE
PLT-1484 Changed first preference load to be done synchronously

### PR DESCRIPTION
**Pull request by hmhealey**

### CODE REVIEW

**Comment by coreyhulen** [comment]
@hmhealey can you base these changes on https://github.com/mattermost/platform/pull/1976

File: web/templates/channel.html [hunk 1]
```diff
@@ -6,7 +6,7 @@
 <body>
   <div id="channel_view" class='channel-view'></div>
 <script>
-window.setup_channel_page({{ .Props }}, {{ .Team }}, {{ .Channel }}, {{ .User }});
+  window.setup_channel_page({{ .Props }}, {{ .Team }}, {{ .Channel }}, {{ .Preferences }});
```
Comment (jwilander) [inline web/templates/channel.html]: I thought removing `{{.User}}` was a mistake but it looks like it wasn't being used before anyways, despite being passed in

**Comment by jwilander** [comment]
+1

**Comment by coreyhulen** [comment]
actually I think it should be in head like https://github.com/mattermost/platform/blob/master/web/templates/head.html#L71

File: web/templates/channel.html (hunk 1 above)
Comment (hmhealey) [inline web/templates/channel.html]: Yeah I noticed it wasn't being used so I just got rid of it

**Comment by hmhealey** [comment]
@coreyhulen Which part do you think should be in head.html?

**Comment by coreyhulen** [comment]
something like

```javascript
window.mm_preferences = {{ .Preferences }};
```

then in the pref store you can have something like whats used for mm_user
```javascript
getCurrentUser() {
        if (this.getProfiles()[global.window.mm_user.id] == null) {
            this.saveProfile(global.window.mm_user);
        }

        return global.window.mm_user;
    }
```

I think we should be consistent with how all the other resources are loaded.  That way preferences will be loaded for stuff like the admin console too.

**Comment by hmhealey** [comment]
@coreyhulen That's a good point. web.go doesn't load preferences yet for other calls, but it should be easy enough to add to those in the future. Instead of creating a `window.mm_preferences` global variable though, I just called `PreferenceStore.setPreferences` directly in head.html since it was already globally accessible. I wanted to avoid having some weird code path where preferences might be in mm_preferences or they might be in the BrowserStore. Does that work for you?

**Pull request merged**